src/useInterApplicationBusPublish.ts

```diff
@@ -1,25 +1,12 @@
-import {useEffect, useState} from "react";
-
-export default <T>(topic: string, message: T) => {
-    const [success, setSuccess] = useState<boolean | null>(null);
-    const [error, setError] = useState<Error | null>(null);
-
-    const onSuccess = () => setSuccess(true);
-    const onFail = (e: Error) => {
-        setError(e);
-        setSuccess(false);
-    };
-
-    useEffect(() => {
+export default <T>(topic: string) => {
+    const publishMessage = (message: T) => {
```

Comment:
does this work when the topic is changed? I'm not sure whether you need to wrap the setting of `publishMessage` in an effect that has a dependency on `[topic]`

Reply:
consider wrapping in useMemo, as discussed